src/Kmd.Logic.Audit.Client.SerilogAzureEventHubs/SerilogAzureEventHubsAuditClient.cs

```diff
@@ -1,39 +1,26 @@
 ﻿using System;
-using System.Net.Http;
-using Serilog;
 using Serilog.Core;
 
 namespace Kmd.Logic.Audit.Client.SerilogAzureEventHubs
 {
     public class SerilogAzureEventHubsAuditClient : IAudit, IDisposable
     {
-        private readonly Logger disposableLogger;
+        private readonly Logger logger;
+        private readonly bool disposeLogger;
         private readonly SerilogLoggerAudit audit;
 
         public SerilogAzureEventHubsAuditClient(SerilogAzureEventHubsAuditClientConfiguration config)
         {
-            if (config == null)
-            {
-                throw new ArgumentNullException(nameof(config));
-            }
-
-            var configBuilder = new LoggerConfiguration()
-                .Enrich.With(new EventIdEnricher())
-                .Enrich.With(new CreatedDateTimeEnricher())
-                .Enrich.WithProperty("_EventSource", config.EventSource)
-                .AuditTo.AzureEventHub(
-                    formatter: new Serilog.Formatting.Compact.CompactJsonFormatter(),
-                    connectionString: config.ConnectionString,
-                    eventHubName: config.AuditEventTopic)
-                ;
-
-            if (config.EnrichFromLogContext == true)
-            {
-                configBuilder = configBuilder.Enrich.FromLogContext();
-            }
+            this.logger = config.DefaultConfiguration().CreateLogger();
+            this.disposeLogger = true;
+            this.audit = new SerilogLoggerAudit(this.logger);
+        }
 
-            this.disposableLogger = configBuilder.CreateLogger();
-            this.audit = new SerilogLoggerAudit(this.disposableLogger);
+        public SerilogAzureEventHubsAuditClient(Logger logger, bool disposeLogger = true)
```

Comment:
We are adding a new constructor on an object that previously had a single constructor. The type can be activated using reflection and DI container.

For example: https://github.com/kmdlogic/kmd-logic-audit-client/blob/76941adfa5b6ad7a8ad87ed79b950856db7b7bea/sample/Kmd.Logic.Audit.Client.AspnetCoreSample/Startup.cs#L35

This instructs `Microsoft.Extensions.DependencyInjection` to create the instance and automatically satisfy the dependencies.

If we add a new ctor with more parameters, I think the one wiht most # of arguments will be chosen as the default by many frameworks.

Reply:
https://docs.microsoft.com/en-us/dotnet/api/microsoft.extensions.dependencyinjection.activatorutilitiesconstructorattribute?view=dotnet-plat-ext-3.1

Reply:
Added as a static method with private constructor instead